tests/test_sqlite_storage.py

```diff
@@ -60,3 +64,130 @@ def test_import_export(temp_dir):
             metrics_after[proj][run] = SQLiteStorage.get_logs(proj, run)
 
     assert metrics_before == metrics_after
+
+
+def _worker_using_sqlite_storage(
+    temp_trackio_dir, project, worker_id, duration_seconds=2, sync_start_time=None
+):
+    """
+    Worker that uses SQLiteStorage methods for database access.
+    This will be protected by ProcessLock when available.
+    """
+    import os
+    import random
+    import sqlite3
+    import time
+
+    os.environ["XDG_CACHE_HOME"] = str(temp_trackio_dir)
+
+    def aggressive_get_connection(db_path):
+        import sqlite3
+
+        conn = sqlite3.connect(str(db_path), timeout=0.01)
+        conn.row_factory = sqlite3.Row
+        return conn
+
+    SQLiteStorage._get_connection = aggressive_get_connection
+
+    if sync_start_time:
+        while time.time() < sync_start_time:
+            time.sleep(0.001)
+
+    run_name = f"worker_{worker_id}"
+    operations = 0
+    db_locked_errors = 0
+
+    start_time = time.time()
+    while time.time() - start_time < duration_seconds:
+        try:
+            if random.random() < 0.6:
```

Comment:
It might be beneficial to use a fixed random seed to improve reproducibility across runs e.g. `random.seed(10)`.

Reply:
Actually just got rid of this since we only use `bulk_log` (added docstrings in `log()` to explain why it's still there)